File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int left = 0;
        int right = nums.size() - 1;
        int res = nums[0];
        
        while (left <= right) {
            // if the portion is already sorted
            if (nums[left] < nums[right]) {
                res = min(res, nums[left]);
                break;
            }
            int mid = left + (right - left) / 2;
            // update the min value
            res = min(res, nums[mid]);
            if (nums[mid] < nums[left]) {
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }
        return res;
    }
};

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        res = nums[0]
        left, right = 0, len(nums) - 1
        
        while left <= right:
            # if the portion is already fully sorted
            if nums[left] < nums[right]:
                res = min(res, nums[left])
                break
            mid = left + (right - left) // 2
            # update the min value(=res)
            res = min(res, nums[mid])
            if nums[mid] < nums[left]:
                right = mid - 1
            else: 
                left = mid + 1
        
        return res